kratos/processes/calculate_distance_to_skin_process.cpp

```diff
@@ -194,21 +204,19 @@ namespace Kratos
 	template<std::size_t TDim>
 	double CalculateDistanceToSkinProcess<TDim>::DistancePositionInSpace(const Node<3> &rNode)
 	{
-		typedef Element::GeometryType intersection_geometry_type;
-        typedef std::vector<std::pair<double, intersection_geometry_type*> > intersections_container_type;
-
         const double epsilon = 1e-12;
 		array_1d<double,TDim> distances;
-        intersections_container_type intersections;
+		unsigned int n_ray_pos(0), n_ray_neg(0);
```

Comment:
on line 207 i think you should be using the epsilon you passed in the constructor

Reply:
This is the `epsilon`to decide what is 0 and what is not. It should be different, indeed, smaller, than the one in the constructor.